### PR TITLE
render-core-node-docs-statically

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Description

Rendering core-node docs statically and removed the need for rendering node-red docs
## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: lib/core-node-docs.js [hunk 1]
```diff
@@ -1,27 +0,0 @@
...
-            errorHandler: {
-                warning: function (w) { },
-                error: function (e) { },
-                fatalError: function (e) { console.error(e) }
-            }
-            // As there's a few nodes that have multiple docs in one file, we wrap it to make it
-            // valid XML
-        }).parseFromString("<scripts>" + data.toString() + "</scripts>", 'text/xml');
-
-        return xpath.select(`/scripts/script[starts-with(@data-help-name, "${node.xpath}")]/*`, doc).join(" ")
```
Comment (ZJvandeWeg) [inline lib/core-node-docs.js:22]: This information will no longer be included in the core node docs?

File: src/node-red/core-nodes/common/catch.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
+---
+eleventyNavigation:
+  key: Catch
+  parent : Common
+---
+
+# Node-RED Function Node
```
Comment (ZJvandeWeg) [inline src/node-red/core-nodes/common/catch.md:7]: Why does it say Function in the title?

File: lib/core-node-docs.js (hunk 1 above)
Comment (sumitshinde-84) [reply lib/core-node-docs.js:22]: No, the core-node docs will not fetch the docs from the node-red package now

File: src/node-red/core-nodes/common/catch.md (hunk 2 above)
Comment (sumitshinde-84) [reply src/node-red/core-nodes/common/catch.md:7]: Oh, sorry about that mistake

File: .eleventy.js [hunk 3]
```diff
@@ -98,18 +97,6 @@ module.exports = function(eleventyConfig) {
         new FlowRenderer().renderFlows(JSON.parse(flow${flowId}), { container: document.getElementById('nr-flow-${flowId}') })</script>`
     });
 
-    eleventyConfig.addGlobalData("coreNodesArray", () => {
```
[the remark this answers is not in the document]
Comment (sumitshinde-84) [reply .eleventy.js:101]: This method of rendering documentation doesn't allow for the addition of new documents to the core-node docs; it only permits the inclusion of documents already present in the upstream. Additionally, I want to ask, if we have documentation for all the nodes, why do we need to fetch the information from the upstream?

[the remark this answers is not in the document]
Comment (ZJvandeWeg) [reply .eleventy.js:101]: Which nodes are missing? It seems like the core nodes are very stable?

> Additionally, I want to ask, if we have documentation for all the nodes, why do we need to fetch the information from the upstream?

A schematic approach that's stable across pages creates trust that the content is complete and predictable.

[the remark this answers is not in the document]
Comment (sumitshinde-84) [reply .eleventy.js:101]: You can refer to the UDP node docs or HTTP proxy. However, these documents do not contain the content we have added. If you look at the MQTT or WebSocket docs, each has a single document for two nodes, such as MQTT in and out, and the same for WebSocket. I have tried to add separate docs for both of them, but it doesn't allow me to add files different from the upstream file structure.

[the remark this answers is not in the document]
Comment (Yndira-E) [reply .eleventy.js:101]: > You can refer to the UDP node docs or HTTP proxy. However, these documents do not contain the content we have added. If you look at the MQTT or WebSocket docs, each has a single document for two nodes, such as MQTT in and out, and the same for WebSocket. I have tried to add separate docs for both of them, but it doesn't allow me to add files different from the upstream file structure.

This was addressed in https://github.com/FlowFuse/website/pull/2429

**Pull request closed without merging**